src/Components/Server/src/Circuits/ServerComponentDeserializer.cs

```diff
@@ -270,6 +270,74 @@ private bool IsWellFormedServerComponent(ComponentMarker record)
         return (componentDescriptor, serverComponent);
     }
 
+    public bool TryDeserializeRootComponentOperations(string serializedComponentOperations, out (RootComponentOperation, ComponentDescriptor?)[] operations)
```

Comment:
We should check for the uniqueness of the Update/Delete operations to ensure that there is at most one update/delete per component ID.